src/cloud-api-adaptor/test/e2e/common_suite.go

```diff
@@ -472,10 +472,10 @@ func DoTestPodToServiceCommunication(t *testing.T, e env.Environment, assert Clo
 func DoTestPodsMTLSCommunication(t *testing.T, e env.Environment, assert CloudAssert) {
 	clientPodName := "curl"
 	clientContainerName := "curl"
-	clientImageName := "docker.io/curlimages/curl:8.4.0"
+	clientImageName := "quay.io/curl/curl:8.4.0"
```

Comment:
In some other places in the code we use `quay.io/curl/curl:latest`.  Can we use the same curl image ? Better is to use a version tag than latest to avoid issues with change of command line options etc.

Also can all these images be moved to separate file as constants ? It'll help to override the images if needed.
I understand the scope of this PR is to just switch out of docker hub mirror, but since you are anyway touching these files thought of suggesting it :-) 
This can be done later on as well.

Reply:
Sure, that makes sense. I was trying to keep versions the same incase that was important, but happy to pin them and separating out as constants is good too. Let me mark this as draft until I get a chance to get around to this.

Reply:
@stevenhorsman  are you thinking in add the version in versions.yaml or just a simple constant in code? 

I'm okay with a constant in code, but if you want the other approach then you want to use https://github.com/confidential-containers/cloud-api-adaptor/pull/1891/commits/5169aec42757d17c571181a0fd3dd8a38688563a#diff-a3bd85dcf59fe8d1cf225b0edbbd31d0390a9588a53b44941ad49d14d9f562eb

Reply:
I'm on the fence with this. I can see the merit of having the versions defined in a single place, but it also doesn't feel quite right to me to mix up test dependencies with the built dependencies in the versions file (which is weird as lots of toolchains do it!).

Reply:
May be have a separate versions file for test deps and placed inside the test folder ?

Reply:
Yeah, that sounds like a reasonable plan.

Reply:
yup, I also don't like the mix up as in Kata. Good idea @bpradipt !

Reply:
So I've defrosted this PR and updated it to add a test versions.yaml with the test images and code to read from there. It feels like quite a bit of extra code, for maybe not a tons better user experience. 

The main issue I have with this is that reading from yaml seems to inherently require the need to throw errors and some of our code in common.go for specifying the busybox image doesn't have a testing.T object or any existing errors thrown, so I don't know if you have any suggestions of how to handle this apart from added return errors and error handling to a whole bunch of methods?